src/enableDisableRules.ts

```diff
@@ -36,25 +37,33 @@ export class EnableDisableRulesWalker extends SkippableTokenAwareRuleWalker {
                 return;
             }
 
+            // keep track of the beginning of the line so that we can use it for same line switches
+            if (scanner.getToken() === ts.SyntaxKind.NewLineTrivia) {
```

Comment:
As opposed to tracking this, you could use:

`node.getLineAndCharacterOfPosition` in conjunction with `scanner.getTokenPos`